src/Compilers/Core/Portable/MetadataReader/PEModule.cs

```diff
@@ -139,35 +140,54 @@ internal PEReader PEReaderOpt
             }
         }
 
-        internal unsafe MetadataReader MetadataReader
+        internal MetadataReader MetadataReader
         {
             get
             {
                 if (_lazyMetadataReader == null)
                 {
-                    MetadataReader newReader;
+                    InitializeMetadataReader();
+                }
+                else if (_isDisposed)
```

Comment:
@tmat i agree that's the existing pattern.  But while we're taking the time to harden the code why not just harden it against future scenarios, and other bugs, as well.